torch/_dynamo/eval_frame.py

```diff
@@ -691,138 +696,174 @@ def result_capturing_wrapper(*graph_inputs):
         return result_capturing_wrapper
 
     flat_args, in_spec = pytree.tree_flatten((args, kwargs))
+    # apply constraints to flat_args
+    # First, reconstruct constraints to be a map:[int] view
+    if not constraints:
+        constraints = list()
 
-    remove_from_cache(f)
-    with patch(f"{__name__}.most_recent_backend", None), config.patch(
-        specialize_int=True
-    ):
-        opt_f = optimize_assert(
-            dynamo_normalization_capturing_compiler,
-            hooks=Hooks(guard_export_fn=guard_export_print, guard_fail_fn=None),
-            export=True,
-            dynamic=(tracing_mode == "symbolic"),
-        )(f)
-        # TODO(voz): We may have instances of `f` that mutate inputs, we should track sideffects and reject.
-        result_traced = opt_f(*args, **kwargs)
-    remove_from_cache(f)
+    tensor_to_indices: Dict[torch.Tensor, Set[int]] = dict()
+    try:
+        for constraint in constraints:
+            tensor = constraint[0]
+            index = constraint[1]
+            # TODO(voz): Is tensor defaultdict friendly?
+            if tensor not in tensor_to_indices:
+                assert not torch._dynamo.has_dynamic_dims(
+                    tensor
+                ), "Illegal to export tensor already marked dynamic"
```

Comment:
This makes the implementation here simpler, but it doesn't make any sense from a UX perspective.  Preexisting markings should be ignored, and left undisturbed.

Reply:
Don't use assert for user error. Raise a regular RuntimeError

Reply:
I don't understand the use case for randomly calling export in the middle of something, where there are already marked dims? It just seems like an accidental / error case. That being said, If you feel strongly about this, I am happy to rewrite this to support restoring state.